### PR TITLE
Remove extra leak detector in RefCountLockResource

### PR DESCRIPTION
**Pull request by jiacheliu3**
### What changes are proposed in this pull request?

The `RefCountLockResource#mTracker` field is removed.

### Why are the changes needed?

`RefCountLockResource` extends `LockResource`, so `RefCountLockResource` is already leak tracked. 

In the current implementation, `RefCountLockResource` has two `mTracker` fields, one declared in its own class, the other from `LockResource`. This doubles the memory footprint for leak detection. Each `DefaultResourceLeak` object retains 1.4KB heap, which is a big cost.
image

### Does this PR introduce any user facing changes?

NA

### CODE REVIEW

File: core/common/src/main/java/alluxio/resource/RefCountLockResource.java [hunk 1]
```diff
@@ -63,9 +52,6 @@ public RefCountLockResource(ReentrantReadWriteLock lock, LockMode mode, boolean
   @Override
   public void close() {
     super.close();
```
Comment (jiacheliu3) [inline core/common/src/main/java/alluxio/resource/RefCountLockResource.java:54]: `RefCountLockResource` direct parent is `RWLockResource` which has no `close()`, so I think this calls the un-overriden `close()` from `LockResource`. 

If this gives you a sense of insecurity, we can add a dummy `close()` to `RWLockResource` to make sure
```
@Override
public void close() {super.close();}
```

Comment (jiacheliu3) [reply core/common/src/main/java/alluxio/resource/RefCountLockResource.java:54]: I manually set the lock pool high watermark to very low to trigger lock eviction. But I don't see any complaint on lock leaks. So I think the current approach is fine.

To stay on the safe side, I think adding a dummy `close()` to `RWLockResource` feels the most secure.

Comment (ggezer) [reply core/common/src/main/java/alluxio/resource/RefCountLockResource.java:54]: No need for dummy methods if the behavior is what you defined here.

File: core/common/src/main/java/alluxio/resource/RWLockResource.java [hunk 2]
```diff
@@ -59,4 +59,13 @@ public boolean downgrade() {
...
   }
+
+  /**
+   * This dummy implementation is just to make sure when a subclass of {@link RWLockResource}
+   * calls {@code super.close()}, {@link LockResource#close()} is called.
+   */
+  @Override
+  public void close() {
+    super.close();
+  }
```
Comment (ggezer) [inline core/common/src/main/java/alluxio/resource/RWLockResource.java:70]: I'd assume super calls should chain up all the way. If that's the case, I believe we can just make sure overriders of close() call super.close() and don't bother with dummy code.

**Review by ggezer — APPROVED** [review]

File: core/common/src/main/java/alluxio/resource/RefCountLockResource.java (hunk 1 above)
Comment (ggezer) [reply core/common/src/main/java/alluxio/resource/RefCountLockResource.java:54]: BTW, I don't think lowering the pool size will trigger the leak. The leak was happening when OOM errors was seen. After @ZhuTopher's fixes, I've never seen a report of such leaks. Could still be happening though...

Comment (jiacheliu3) [reply core/common/src/main/java/alluxio/resource/RefCountLockResource.java:54]: Verified the behavior is as we expected, the propagation works:
1. RefCountLockResource.close() calls super.close()
1. RWLockResource does not override close(), so LockResource.close() is actually called. This is what we want.

No need for the dummy method, I will remove that.

**Comment by jiacheliu3** [comment]
Failed test is irrelevant to the change
```
Error: 1.373 [ERROR] Tests run: 10, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 1,002.969 s <<< FAILURE! - in alluxio.server.ft.journal.JournalBackupIntegrationTest
Error: 1.374 [ERROR] alluxio.server.ft.journal.JournalBackupIntegrationTest.backupRestoreEmbedded  Time elapsed: 600.024 s  <<< ERROR!
org.junit.runners.model.TestTimedOutException: test timed out after 600000 milliseconds
	at java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1868)
	at alluxio.grpc.GrpcConnectionPool.acquireConnection(GrpcConnectionPool.java:101)
	at alluxio.grpc.GrpcChannelBuilder.build(GrpcChannelBuilder.java:121)
	at alluxio.master.PollingMasterInquireClient.pingMetaService(PollingMasterInquireClient.java:161)
	at alluxio.master.PollingMasterInquireClient.getAddress(PollingMasterInquireClient.java:135)
	at alluxio.master.PollingMasterInquireClient.getPrimaryRpcAddress(PollingMasterInquireClient.java:110)
	at alluxio.AbstractMasterClient.getAddress(AbstractMasterClient.java:56)
	at alluxio.AbstractClient.connect(AbstractClient.java:217)
	at alluxio.client.file.BaseFileSystem.rpc(BaseFileSystem.java:565)
	at alluxio.client.file.BaseFileSystem.exists(BaseFileSystem.java:200)
	at alluxio.client.file.FileSystem.exists(FileSystem.java:275)
	at alluxio.server.ft.journal.JournalBackupIntegrationTest.backupRestoreTest(JournalBackupIntegrationTest.java:414)
	at alluxio.server.ft.journal.JournalBackupIntegrationTest.backupRestoreEmbedded(JournalBackupIntegrationTest.java:139)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:288)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:282)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)

Error: 1.403 [ERROR] alluxio.server.ft.journal.JournalBackupIntegrationTest.backupRestoreEmbedded  Time elapsed: 87.65 s  <<< ERROR!
alluxio.exception.FileAlreadyExistsException: Not allowed to create directory because path already exists: /dir1
	at alluxio.client.file.BaseFileSystem.rpc(BaseFileSystem.java:570)
	at alluxio.client.file.BaseFileSystem.createDirectory(BaseFileSystem.java:150)
	at alluxio.server.ft.journal.JournalBackupIntegrationTest.backupRestoreTest(JournalBackupIntegrationTest.java:399)
	at alluxio.server.ft.journal.JournalBackupIntegrationTest.backupRestoreEmbedded(JournalBackupIntegrationTest.java:139)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:288)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:282)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**